include/rapidcheck/Assertions.h

```diff
@@ -19,6 +19,36 @@
                                        __FILE__, __LINE__, (description)));    \
   } while (false)
 
+#define RC_INTERNAL_CHECK_THROWS_RESULT(ResultType, expression, description)   \
+  do {                                                                         \
+    try {                                                                      \
+      expression;                                                              \
+    } catch (...) {                                                            \
+      break;                                                                   \
+    }                                                                          \
+    throw ::rc::detail::CaseResult(                                            \
+        ::rc::detail::CaseResult::Type::ResultType,                            \
+        ::rc::detail::makeDescriptionMessage(                                  \
+            __FILE__, __LINE__, description "(" #expression ")"));             \
```

Comment:
I'd really like it to say on failure that it was because no exception was thrown for some reason.